android/app/src/main/java/org/uwaterloo/subletr/pages/home/HomePageViewModel.kt

```diff
@@ -1,91 +1,311 @@
+import org.uwaterloo.subletr.pages.home.map.MAX_DISTANCE_IN_MINUTES
 import org.uwaterloo.subletr.services.INavigationService
+import org.uwaterloo.subletr.utils.UWATERLOO_LATITUDE
+import org.uwaterloo.subletr.utils.UWATERLOO_LONGITUDE
+import org.uwaterloo.subletr.utils.base64ToBitmap
 import javax.inject.Inject
 
 @HiltViewModel
 class HomePageViewModel @Inject constructor(
 	private val navigationService: INavigationService,
 	val homeListChildViewModel: HomeListChildViewModel,
 	val homeMapChildViewModel: HomeMapChildViewModel,
+	private val listingsApi: ListingsApi,
 ) : SubletrViewModel<HomePageUiState>(
 	homeListChildViewModel,
 	homeMapChildViewModel,
 ) {
 	val navHostController: NavHostController get() = navigationService.navHostController
 
-	private val internalMergedUiStateStream = Observable.merge(
-		homeListChildViewModel.uiStateStream,
-		homeMapChildViewModel.uiStateStream
+	private val totalNumberOfPagesStream: BehaviorSubject<Int> =
+		BehaviorSubject.createDefault(1)
+
+	private val getListingParamsStream: Observable<GetListingParams> = BehaviorSubject.createDefault(
+		GetListingParams(
+			filters = HomePageUiState.FiltersModel(
+				locationRange = HomePageUiState.LocationRange(),
+				priceRange = HomePageUiState.PriceRange(),
+				roomRange = HomePageUiState.RoomRange(),
+				gender = Gender.OTHER,
+				housingType = HousingType.OTHER,
+				dateRange = HomePageUiState.DateRange(),
+				favourite = false,
+				timeToDestination = null,
+			),
+			transportationMethod = HomePageUiState.TransportationMethod.ALL,
+			homePageView = HomePageUiState.HomePageViewType.LIST,
+		)
+	)
+		.distinctUntilChanged()
+
+	private val listingsStream: Observable<ListingParamsAndResponse> = Observable.merge(
+		getListingParamsStream,
+		homeListChildViewModel.getListingParamsStream,
+		homeMapChildViewModel.getListingParamsStream,
 	)
+		.distinctUntilChanged { first, second ->
+			first.listingPagingParams.pageNumber == second.listingPagingParams.pageNumber &&
+				first.filters.deepEquals(second.filters) &&
+				first.transportationMethod == second.transportationMethod
+		}
 		.map {
-			internalPublishUiStateStream.onNext(it)
+			if (it.listingPagingParams.pageNumber == 0) {
+				it.copy(
+					listingPagingParams = it.listingPagingParams.copy(
+						previousListingItemsModel = HomePageUiState.ListingItemsModel(
+							listings = emptyList(),
+							likedListings = emptySet(),
+							listingsImages = emptyList(),
+						)
+					)
+				)
+			}
+			else {
+				it
+			}
 		}
+		.map { getListingParams: GetListingParams ->
+			ListingParamsAndResultResponse(
+				listingParams = getListingParams,
+				listingsResponse = runCatching {
+					runBlocking {
+						// TODO: Change to use filter values + latitude + longitude
+						listingsApi.listingsList(
+							longitude = UWATERLOO_LONGITUDE,
+							latitude = UWATERLOO_LATITUDE,
+							pageNumber = getListingParams.listingPagingParams.pageNumber,
```

Comment:
Just curious, how is this gonna work on the map page? like are we gonna sort the posts by distance?

Reply:
Backend will have to do that lol
We can't sort on frontend because pagination